community/kernel/src/main/java/org/neo4j/GraphDatabaseBuilder.java

```diff
@@ -0,0 +1,384 @@
+    {
+        params.put( GraphDatabaseSettings.lucene_searcher_cache_size.name(), String.valueOf( count ) );
+        return self();
+    }
+
+    /**
+     * Set the threshold, in days, after which logical logs will be pruned.
+     *
+     * @param thresholdDays the threshold, in days, after which logical logs will be pruned
+     * @return this builder
+     */
+    public BUILDER withLogicalLogPruningThresholdDays( int thresholdDays )
+    {
+        params.put( GraphDatabaseSettings.keep_logical_logs.name(), String.valueOf( thresholdDays ) + " days" );
+        return self();
+    }
+
+    /**
+     * Set the threshold, in bytes, after which logical logs will be pruned.
+     *
+     * @param thresholdSize the threshold, in bytes, after which logical logs will be pruned
+     * @return this builder
+     */
+    public BUILDER withLogicalLogPruningThresholdSize( long thresholdSize )
+    {
+        params.put( GraphDatabaseSettings.keep_logical_logs.name(), String.valueOf( thresholdSize / 1024 ) + " k" );
+        return self();
+    }
+
+    /**
+     * Set the maximum number of archived logical logs.
+     *
+     * @param count the maximum number of archived logical logs
+     * @return this builder
+     */
+    public BUILDER withLogicalLogMaxArchives( int count )
+    {
+        params.put( GraphDatabaseSettings.keep_logical_logs.name(), String.valueOf( count ) + " files" );
+        return self();
+    }
+
+    /**
+     * Set the threshold, in bytes, for logical log rotation.
+     *
+     * @param threshold the threshold, in bytes, for logical log rotation
+     * @return this builder
+     */
+    public BUILDER withLogicalLogRotationThreshold( long threshold )
+    {
+        params.put( GraphDatabaseSettings.logical_log_rotation_threshold.name(), String.valueOf( threshold ) );
+        return self();
+    }
+
+    /**
+     * The amount of memory to use for mapping the store files, in bytes. If Neo4j is running on a dedicated server,
+     * then it is generally recommended to leave about 2-4 gigabytes for the operating system, give the
+     * JVM enough heap to hold all your transaction state and query context, and then leave the rest for
+     * the page cache. If this is not set, then it is assumed that Neo4j is running exclusively on the machine,
+     * and the size will be heuristically set to 75% of RAM minus the max Java heap size.
+     *
+     * @param size the amount of memory to use for mapping the store files, in bytes
+     * @return this builder
+     */
+    public BUILDER usingPageCacheMemory( long size )
+    {
+        params.put( GraphDatabaseSettings.pagecache_memory.name(), String.valueOf( size ) );
+        return self();
+    }
+
+    /**
+     * Set an internal parameter. Use of this method is not recommended.
+     *
+     * @param key parameter key
+     * @param value parameter value
+     * @return this builder
+     */
+    public BUILDER withParam( String key, String value )
```

Comment:
I'm happy to call it `withConfig`. However, I'm also thinking it should be as obtuse as possible - since it really shouldn't be used by the end-user, and is only exposed as a transitional thing (indeed, now I'm wondering if we could/should only expose it in the `TestGraphDatabase`?).